shared/src/main/scala/coop/rchain/shared/RChainEvent.scala

```diff
@@ -0,0 +1,61 @@
+package coop.rchain.shared
+
+import cats.Applicative
+
+sealed trait RChainEvent {}
+
+final case class BlockCreated(
+    blockHash: String,
+    parentHashes: List[String],
+    justificationHashes: List[(String, String)],
+    deployIds: List[String],
+    creator: String,
+    seqNum: Int
+) extends RChainEvent
+
+final case class BlockAdded(
+    blockHash: String,
+    parentHashes: List[String],
+    justificationHashes: List[(String, String)],
+    deployIds: List[String],
+    creator: String,
+    seqNum: Int
+) extends RChainEvent
+
+final case class BlockFinalised(blockHash: String) extends RChainEvent
+
+object RChainEvent {
+  def created(
+      bs: String,
+      parents: List[String],
+      justifications: List[(String, String)],
+      deployIds: List[String],
+      creator: String,
+      seqNum: Int
+  ): RChainEvent =
+    BlockCreated(bs, parents, justifications, deployIds, creator, seqNum)
+
+  def added(
+      bs: String,
+      parents: List[String],
+      justifications: List[(String, String)],
+      deployIds: List[String],
+      creator: String,
+      seqNum: Int
+  ): RChainEvent =
+    BlockAdded(bs, parents, justifications, deployIds, creator, seqNum)
+
+  def finalised(bs: => String): RChainEvent = BlockFinalised(bs)
+}
+
+trait EventPublisher[F[_]] {
+  def publish(e: => RChainEvent): F[Unit]
+}
```

Comment:
is there a need/gain in that?

Reply:
I feel that this will be heavily bolted on. I'm fine with ticketing a design effort to do such a step, but I don't see how to deliver it as part of this PR.

Reply:
Ok, let's postpone this discussion (with the SRE team)

Reply:
https://rchain.atlassian.net/browse/RCHAIN-3837